helper.php

```diff
@@ -593,6 +538,84 @@ function _permalink(&$ins, $page, $sect, $flags) {
         $ins[1] = array('include_header', array($ins[1][0], $ins[1][1], $ins[1][2], $page, $sect, $flags));
     }
 
+    /**
+     * Convert internal and local links depending on the included pages
+     *
+     * @param array  $ins            The instructions that shall be adapted
+     * @param string $page           The included page
+     * @param array  $included_pages The array of pages that are included
+     */
+    private function adapt_links(&$ins, $page, $included_pages = null) {
+        $num = count($ins);
+        $ns  = getNS($page);
+
+        for($i=0; $i<$num; $i++) {
+            // adjust links with image titles
+            if (strpos($ins[$i][0], 'link') !== false && isset($ins[$i][1][1]) && is_array($ins[$i][1][1]) && $ins[$i][1][1]['type'] == 'internalmedia') {
+                // resolve relative ids, but without cleaning in order to preserve the name
+                $media_id = resolve_id($ns, $ins[$i][1][1]['src']);
+                // make sure that after resolving the link again it will be the same link
+                if ($media_id{0} != ':') $media_id = ':'.$media_id;
+                $ins[$i][1][1]['src'] = $media_id;
+            }
+            switch($ins[$i][0]) {
+                case 'internallink':
+                case 'internalmedia':
+                    // make sure parameters aren't touched
+                    $link_params = '';
+                    $link_id = $ins[$i][1][0];
+                    $link_parts = explode('?', $link_id, 2);
+                    if (count($link_parts) === 2) {
```

Comment:
The [PHP documentation on arrays](http://php.net/manual/en/language.types.array.php) says 

> Attempting to access an array key which has not been defined is the same as accessing any other undefined variable: an E_NOTICE-level error message will be issued, and the result will be NULL.

I would like to avoid triggering notices.